providers/openfeature-provider-flagd/tests/e2e/test_config.py

```diff
@@ -0,0 +1,90 @@
+import re
+import sys
+
+import pytest
+from asserts import assert_equal
+from pytest_bdd import parsers, scenarios, then, when
+
+from openfeature.contrib.provider.flagd.config import CacheType, Config, ResolverType
+
+
+def camel_to_snake(name):
+    name = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", name)
+    return re.sub("([a-z0-9])([A-Z])", r"\1_\2", name).lower()
+
+
+def str2bool(v):
+    return v.lower() in ("yes", "true", "t", "1")
+
+
+type_cast = {
+    "Integer": int,
+    "String": str,
+    "Boolean": str2bool,
+    "ResolverType": ResolverType,
+    "CacheType": CacheType,
+}
+
+
+@pytest.fixture(autouse=True, scope="module")
+def setup(request):
+    pass
+
+
+@pytest.fixture()
+def option_values() -> dict:
+    return {}
+
+
+@when(
+    parsers.cfparse(
+        'we have an option "{option}" of type "{type_info}" with value "{value}"',
+    ),
+)
+def option_with_value(option: str, value: str, type_info: str, option_values: dict):
+    value = type_cast[type_info](value)
+    option_values[camel_to_snake(option)] = value
+
+
+@when(
+    parsers.cfparse(
+        'we have an environment variable "{env}" with value "{value}"',
+    ),
+)
+def env_with_value(monkeypatch, env: str, value: str):
+    monkeypatch.setenv(env, value)
+
+
+@when(
+    parsers.cfparse(
+        "we initialize a config",
+    ),
+    target_fixture="config",
+)
+def initialize_config(option_values):
+    return Config(**option_values)
+
+
+@when(
+    parsers.cfparse(
+        'we initialize a config for "{resolver_type}"',
+    ),
+    target_fixture="config",
+)
+def initialize_config_for(resolver_type: str, option_values: dict):
+    return Config(resolver_type=ResolverType(resolver_type), **option_values)
+
+
+@then(
+    parsers.cfparse(
+        'the option "{option}" of type "{type_info}" should have the value "{value}"',
+    )
+)
+def check_option_value(option, value, type_info, config):
+    value = type_cast[type_info](value)
+    value = value if value != "null" else None
+    assert_equal(config.__getattribute__(camel_to_snake(option)), value)
+
+
+if sys.version_info >= (3, 9):
```

Comment:
this does not work on python 3.8

Reply:
Why isn't Python 3.8 supported? What will happen if someone uses that version?

Reply:
It is just for this test - somehow the markers don't work with python 3.8, hence i can not load this scenario, as it is not filtering properly